src/core/tests/conftest.py

```diff
@@ -182,64 +182,41 @@ def auth_header_user_permissions(access_token_user_permissions):
     for item in items:
-        if e2e_test not in item.keywords:
-            item.add_marker(skip_non_e2e)
+        if keyword not in item.keywords:
+            item.add_marker(skip_marker)
 
 
-def skip_for_e2e_admin(items):
-    skip_non_doc_pictures = pytest.mark.skip(reason="need --e2e-admin option to run tests marked with e2e_admin")
-    for item in items:
-        if "e2e_admin" not in item.keywords:
-            item.add_marker(skip_non_doc_pictures)
+def pytest_collection_modifyitems(config, items):
+    options = {
+        "--e2e-ci": ("e2e_ci", "skip for --e2e-ci test"),
+        "--e2e-user": ("e2e_user", "skip for --e2e-user test"),
+        "--e2e-admin": ("e2e_admin", "need --e2e-admin option to run tests marked with e2e_admin"),
+        "--e2e-user-workflow": ("e2e_user_workflow", "need --e2e-user-workflow option to run tests marked with e2e_user_workflow"),
+    }
 
+    config.option.start_live_server = False
+    config.option.headed = True
 
-def skip_for_e2e_user_workflow(items):
-    skip_non_doc_pictures = pytest.mark.skip(reason="need --e2e-user-workflow option to run tests marked with e2e_user_workflow")
-    for item in items:
-        if "e2e_user_workflow" not in item.keywords:
-            item.add_marker(skip_non_doc_pictures)
 
+    for option, (keyword, reason) in options.items():
+        if config.getoption(option):
+            if option == "--e2e-ci":
+                config.option.headed = False
+            skip_tests(items, keyword, reason)
+            return
 
-def pytest_collection_modifyitems(config, items):
-    e2e_type = config.getoption("--e2e-user-ci") or config.getoption("--e2e-user")
-    e2e_admin = config.getoption("--e2e-admin")
-    e2e_user_workflow = config.getoption("--e2e-user-workflow")
-
-    if e2e_type:
-        config.option.start_live_server = False
-        config.option.headed = e2e_type == "e2e_user"
-        skip_for_e2e(e2e_type, items)
-        return
-
-    if e2e_admin:
-        config.option.start_live_server = False
-        config.option.headed = True
-        skip_for_e2e_admin(items)
-        return
-
-    if e2e_user_workflow:
-        config.option.start_live_server = False
-        config.option.headed = True
-        skip_for_e2e_user_workflow(items)
-        return
-
-    # Skip all e2e and e2e_admin tests if no relevant flag is provided
-    skip_all = pytest.mark.skip(reason="need --e2e-user, --e2e-user-ci, --e2e-user-workflow, or --e2e-admin option to run these tests")
+
+    skip_all = pytest.mark.skip(reason="need --e2e-user, --e2e-ci, --e2e-user-workflow, or --e2e-admin option to run these tests")
     for item in items:
-        if (
-            "e2e_user" in item.keywords
-            or "e2e_user_ci" in item.keywords
-            or "e2e_admin" in item.keywords
-            or "e2e_user_workflow" in item.keywords
-        ):
+        if any(keyword in item.keywords for keyword in options.values()):
             item.add_marker(skip_all)
```

Comment:
If there is an issue with pytest running e2e without --e2e flag present it is probably in this logic or some issue above this so this part is never actually executed